src/platform/ti/cc13xx_26xx/BLEManagerImpl.cpp

```diff
@@ -215,25 +327,73 @@ uint16_t BLEManagerImpl::_NumConnections(void)
     return numConns;
 }
 
+/*
+Iterate through all available connections via connIndex.
+Connection is valid if handle is not NULL.
+*/
+BLE_CONNECTION_OBJECT BLEManagerImpl::GetConnection(uint8_t connIndex)
+{
+    BLE_CONNECTION_OBJECT conId = NULL;
+    if (connIndex < MAX_NUM_BLE_CONNS)
+    {
+        conId = (BLE_CONNECTION_OBJECT) &connList[connIndex].connHandle;
+    }
+    return (conId);
+}
+
+ConnectivityManager::CHIPoBLEServiceMode BLEManagerImpl::getMatteroBLEServiceMode(BLE_CONNECTION_OBJECT conId)
+{
+    ConnectivityManager::CHIPoBLEServiceMode mServiceMode = ConnectivityManager::kCHIPoBLEServiceMode_NotSupported;
+    uint8_t i;
+    for (i = 0; i < MAX_NUM_BLE_CONNS; i++)
+    {
+        if (connList[i].mServiceMode == ConnectivityManager::kCHIPoBLEServiceMode_Enabled)
+        {
+            mServiceMode = connList[i].mServiceMode;
+            break;
+        }
+    }
+    return mServiceMode;
+}
```

Comment:
It looks like conId is not being used, is that intentional?

Reply:
This could be improved:
```
ConnectivityManager::CHIPoBLEServiceMode BLEManagerImpl::getMatteroBLEServiceMode(BLE_CONNECTION_OBJECT conId)
{
    ConnectivityManager::CHIPoBLEServiceMode mServiceMode = ConnectivityManager::kCHIPoBLEServiceMode_NotSupported;
    uint8_t index;

    index = GetBLEConnIndex(*((uint32_t *) conId));

    if (index < MAX_NUM_BLE_CONNS)
    {
        mServiceMode = connList[index].mServiceMode
    }

    return mServiceMode;
}
```